### PR TITLE
DOCSP-39852: Fluent API

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-39852
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/rust/DOCSP-39852-fluent-api/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/gridfs.txt [hunk 1]
```diff
@@ -113,19 +113,18 @@ creates a reference to either a new or existing GridFS bucket:
...
+``GridFsBucketOptions`` struct.
 
-.. note:: Instantiating Options
+.. note:: Instantiating Structs
    
    The {+driver-short+} implements the Builder design pattern for the
-   creation of many different types, including ``GridFsBucketOptions``. You
-   can use the ``builder()`` method to construct an instance of each type
+   creation of some struct types, including ``GridFsBucketOptions``. You can
+   use the ``builder()`` method to construct an instance of each type
```
Comment (rustagir) [inline source/fundamentals/gridfs.txt:123]: Q: just checking that the gridFS api is still using builders to create options

File: source/includes/fundamentals/code-snippets/quick-reference-async.rs [hunk 2]
```diff
@@ -135,57 +114,43 @@ async fn main() -> mongodb::error::Result<()> {
...
     // end-sort
 
     // start-project
-    let opts: FindOptions = FindOptions::builder()
-        .projection(doc! { "title": 1, "metacritic": 1, "_id": 0 })
-        .build();
-
     let filter = doc! { "year": 2015 };
-    let mut cursor = collection.find(filter, opts).await?;
+    let mut cursor = collection
```
Comment (rustagir) [inline source/includes/fundamentals/code-snippets/quick-reference-async.rs:143]: Q: with all these line changes, can you check that theres no code highlighting that was affected?

**Review by rustagir — APPROVED** [review]
Looks great, awesome work on this hefty PR - lmk if you want me to take another look after fixing small errors

File: source/fundamentals/crud/compound-operations.txt [hunk 3]
```diff
@@ -175,13 +174,16 @@ The following table describes the options available in
 
 .. TODO add links to guides for relevant options
 
-The {+driver-short+} implements the Builder design pattern for the
-creation of a ``FindOneAndDeleteOptions`` instance. You can use the
-type's ``builder()`` method to construct an options instance by
-chaining option builder functions one at a time.
+.. note:: Setting Options
+   
+   Beginning in {+driver-short+} v3.0, you can set ``FineOneAndDeleteOptions`` fields by
```
[the remark this answers is not in the document]
Comment (norareidy) [reply source/fundamentals/crud/compound-operations.txt:179]: oops fixed!

File: source/fundamentals/gridfs.txt (hunk 1 above)
Comment (norareidy) [reply source/fundamentals/gridfs.txt:123]: Yes it is

File: source/faq.txt [hunk 4]
```diff
@@ -134,7 +134,7 @@ and using the ``try_next()`` method to iterate over ``Actor`` instances causes a
    // Add setup code here 
 
    let my_coll: Collection<Actor> = client.database("db").collection("actors");
-   let mut cursor = my_coll.find(None, None).await?;
+   let mut cursor = my_coll.find(None).await?;
```
Comment (abr-egn) [inline source/faq.txt:137]:
```suggestion
   let mut cursor = my_coll.find(doc! {}).await?;
```

File: source/includes/fundamentals/code-snippets/quick-reference-sync.rs [hunk 5]
```diff
@@ -15,24 +15,21 @@ fn main() -> Result<()> {
     let collection: Collection<Document> = client.database("sample_mflix").collection("movies");
 
     // start-find-one
-    let result = collection.find_one(
-        doc! { "title": "Peter Pan" },
-        None
-    )?;
+    let result = collection.find_one(doc! { "title": "Peter Pan" })?;
```
Comment (abr-egn) [inline source/includes/fundamentals/code-snippets/quick-reference-sync.rs:18]: Unfortunately, for the sync API the switch to fluent methods means they now need a `.run()` to execute; there's no natural terminator like `.await` is for the async API.  So this becomes:
```
let result = collection.find_one(doc! { "title": "Peter Pan" }).run()?;
```
and similar for other places below.

**Review by abr-egn — APPROVED** [review]
LGTM!

**Pull request merged**